worker-workflow/src/main/java/com/github/cafdataprocessing/workflow/ArgumentsManager.java

```diff
@@ -62,27 +69,76 @@ public ArgumentsManager(final SettingsApi settingsApi, final String settingsServ
         }
         final Cache cache = new Cache(settingsServiceCacheDirectory, SETTINGS_SERVICE_CACHE_SIZE_BYTES);
         okHttpClient.setCache(cache);
-        okHttpClient.networkInterceptors().add(getCacheControlInterceptor());
+        okHttpClient.networkInterceptors().add(new SetCacheMaxAgeInterceptor());
+        okHttpClient.networkInterceptors().add(new RecordLastAccessTimeInterceptor());
         final ApiClient apiClient = new ApiClient();
         apiClient.setBasePath(settingsServiceUrl);
         apiClient.setHttpClient(okHttpClient);
         settingsApi.setApiClient(apiClient);
+        forceCacheRefreshInterceptor = new ForceCacheRefreshInterceptor();
+        settingsServiceLastAccessTimeMap
+            = ExpiringMap
+                .builder()
+                .expiration(SETTINGS_SERVICE_CACHE_EXPIRATION_TIME_MINUTES, TimeUnit.MINUTES)
+                .build();
     }
 
-    private Interceptor getCacheControlInterceptor() {
-        return chain -> {
-            final Response response = chain.proceed(chain.request());
-            final CacheControl cacheControl = new CacheControl.Builder()
-                    .maxAge(5, TimeUnit.MINUTES)
+    private final class SetCacheMaxAgeInterceptor implements Interceptor
```

Comment:
I think this class can be `static`.
```suggestion
    private static final class SetCacheMaxAgeInterceptor implements Interceptor
```